### DISCUSSION TITLE
Is it possible to download the Quarto file in html and run the interactive plot locally without a hosting server? Thanks a lot!

### DISCUSSION THREAD

**Discussion by hmeng9521** [quarto-dev/quarto-cli#10908] [Q&A]
### Description

Hi,

Thanks a lot for this powerful tool, I feel it can help in many fields of usage.
I wanted to create an interactive plot using Observable JS, and see if it can be sent to reviewers for review, provided that they don't need any application installed on their end or visiting a website. 
Is it possible to create an standalone html file with Observable JS and Quarto, and send to end user while the interactivity still works, and no need for application running in the background/server connection needed?

Reason is that the data is a bit confidential, so we try to avoid hosting app or uploading data outside the organization.

Thanks!

**Comment by cscheid** [comment]
Unfortunately it's not, and it's not something we can control. OJS uses JS features that are not available in `file://` schemes. With that said, you can send the entire rendered project and give them instructions to run it on a local server. If they have Python installed that's simply a matter of calling `python -m http.serve` from the terminal.

**Comment by hmeng9521** [comment]
Got it, thanks so much for the quick answer.
I wish this powerful tool to be applied in many more use cases!


Carlos Scheidegger ***@***.***>于2024年9月27日 周五10:57写道：

**Reply by mcanouil** [reply]
That's not up to Quarto to change how web technologies such as HTML and JavaScript work.
Quarto is only using those tools thus it has the same limitations.

**Comment by hmeng9521** [comment]
I see, that makes sense. Thanks for the reply!

Mickaël Canouil ***@***.***> 于2024年9月29日周日 11:30写道：

**Comment by GuillaumeChretienCerema** [comment]
Hi there,
I’m not sure if this is the most appropriate solution, but it works perfectly for my needs with just three files :

- caddy.exe [cf caddy repositery](https://github.com/caddyserver/caddy) and the builded istribution in [release](https://github.com/caddyserver/caddy/releases/tag/v2.10.2)
- A Caddyfile (no extension) with a configuration like this:

```
:8081 {
    # racine relative
    root * ./_site

    # serveur de fichiers par défaut pour les autres routes
    file_server
}
```

- A launch.bat script to start everything:
```
@echo off
cd /d "%~dp0"

REM --- Lancer Caddy en arrière-plan ---
start "" /B caddy.exe run --config Caddyfile

REM --- Attendre une seconde pour que le serveur démarre ---
timeout /t 1 >nul

REM --- Ouvrir le navigateur sur localhost ---
start "" http://localhost:8081

pause
```

How I Use It :
I render my website with Quarto and simply double-click launch.bat. It even handles HTTPS range requests (which is essential for my use case with pmtiles format). All my ObservableJavaScript chunks and website features (like search) work flawlessly. This setup lets me run a production-like environment directly from my local disk—no server required!

Note: I'm aware this approach may raise security concerns and that a proper server solution is still recommended for public-facing projects. If this kind of solution seems problematic to you or could be better configured, I’d love to hear your thoughts or insights !
